JMS/com/ibm/mq/samples/jms/ConversationSharingTest.java

```diff
@@ -0,0 +1,154 @@
+/*
+* (c) Copyright IBM Corporation 2024
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package com.ibm.mq.samples.jms;
+
+import javax.jms.JMSContext;
+import javax.jms.JMSException;
+
+// Use these imports for building with Jakarta Messaging
+// import jakarta.jms.JMSContext;
+// import jakarta.jms.JMSException;
+
+import com.ibm.msg.client.jms.JmsConnectionFactory;
+import com.ibm.msg.client.jms.JmsFactoryFactory;
+import com.ibm.msg.client.wmq.WMQConstants;
+
+public class ConversationSharingTest {
+
+  // System exit status value (assume unset value to be 1)
+  private static int status = 1;
+
+  // Create variables for the connection to MQ
+  private static final String HOST = "localhost"; // Host name or IP address
+  private static final int PORT = 1414; // Listener port for your queue manager
+  private static final String CHANNEL = "JMS.SVRCONN"; // Channel name
+  private static final String QMGR = "QM1"; // Queue manager name
+  private static final String APP_USER = ""; // User name that application uses to connect to MQ
+  private static final String APP_PASSWORD = ""; // Password that the application uses to connect to MQ
+
+  /**
+   * Main method
+   *
+   * @param args
+   */
+  public static void main(String[] args) {
+
+    // Variables
+    JMSContext context1 = null;
+    JMSContext context2 = null;
+
+    try {
+      // Create a connection factory
+      JmsFactoryFactory ff = JmsFactoryFactory.getInstance(WMQConstants.WMQ_PROVIDER);
+      JmsConnectionFactory cf = ff.createConnectionFactory();
+
+      // Set the properties
+      cf.setStringProperty(WMQConstants.WMQ_HOST_NAME, HOST);
```

Comment:
These hard coded configuration settings, will need to be flexible to allow them to be overridden, by at least environment settings.